sklearn/neighbors/_classification.py

```diff
@@ -69,6 +69,9 @@ class KNeighborsClassifier(KNeighborsMixin, ClassifierMixin, NeighborsBase):
         minkowski, and with p=2 is equivalent to the standard Euclidean
         metric. For a list of available metrics, see the documentation of
         :class:`~sklearn.metrics.DistanceMetric`.
+        Metrics listed in metrics.pairwise.PAIRWISE_DISTANCE_FUNCTIONS are
+        also supported. For the 'cosine' metric, the
```

Comment:
Looking at the docstring as a whole, I think we can combine the two sentences:

```
For a list of available metrics, see the documentation of
:class:`~sklearn.metrics.DistanceMetric` and the metrics listed in
`metrics.pairwise.PAIRWISE_DISTANCE_FUNCTIONS`. Note that the
"cosine" metric uses :func:`~sklearn.metrics.pairwise.cosine_distances`.
```

(Note that the backticks around `~sklearn.metrics.pairwise.cosine_distances` are required to get the link to work)